### PR TITLE
Feature addition: Wizcli Parser

### PR DESCRIPTION
**Pull request by OsamaMahmood**
**Description**

This PR is related to new parser being added to Defectdojo. To my knowledge and reading the documentation related to different parsers supported by DD i was not able to find parser for Wizcli which is a command line utility provided by https://wiz.io as part of there platform. That motivated me to work on the parser as it was something that we also wanted to use with in our org.

**Checklist**

This checklist is for your information.

- [x] Make sure to rebase your PR against the very latest `dev`.
- [x] Features/Changes should be submitted against the `dev`.
- [x] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [x] Your code is flake8 compliant.
- [x] Your code is python 3.11 compliant.
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [x] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR.

### CODE REVIEW

**Comment by mtesauro** [comment]
@OsamaMahmood I've kicked off the GH Actions on this PR but you'll need to fix the ruff linter issues before this will be reviewed closely for approval.

Thanks for the contribution and best of luck.

**Comment by OsamaMahmood** [comment]
Thanks @mtesauro will be fixing the ruff-linting issues

**Comment by OsamaMahmood** [comment]
@mtesauro all issues resolved removed code duplication and fix ruff linting issues.

**Comment by mtesauro** [comment]
@OsamaMahmood Just re-kicked the tests, I'll look again once those have run.

**Review by mtesauro — APPROVED** [review]
Approved

**Comment by OsamaMahmood** [comment]
Thanks @mtesauro for the approval looking forward to the merge.

**Comment by OsamaMahmood** [comment]
hey team (@Maffooch @manuel-sommer @hblankenship) can you help me get this PR reviewed @mtesauro has review and approved these changes need 3 more review on this before it could be merged to `dev` . If you don't merge it, merge conflicts will start to accumulate.

File: dojo/fixtures/test_type.json
Comment (Maffooch) [inline dojo/fixtures/test_type.json:70]: This fixture is for test types that do not have a parser associated with them. The new test types you are adding here will be generated dynamically
```suggestion
    }
```

**Review by Maffooch — CHANGES_REQUESTED** [review]
Everything else looks good besides this one thing

File: dojo/fixtures/test_type.json
Comment (OsamaMahmood) [reply dojo/fixtures/test_type.json]: done @Maffooch

**Review by Maffooch — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Review by hblankenship — APPROVED** [review]

**Pull request merged**
